redaxo/src/core/lib/fragment.php

```diff
@@ -1,6 +1,13 @@
 <?php
 
 /**
+ * Methods declared here are available to be called from within a fragment-file, even if not public:.
+ *
+ * @method void   subfragment(string $filename, array $params = [])
+ * @method string getSubfragment(string $filename, array $params = [])
+ * @method string i18n(string $key, ...$replacements)
+ * @method mixed  escape($value, $strategy = 'html')
+ *
```

Comment:
Gibt es noch eine andere Variante das zu definieren? Oder müsste PHPStorm hier was nachrüsten?

<img width="734" alt="Bildschirmfoto 2020-11-19 um 16 07 38" src="https://user-images.githubusercontent.com/338267/99684308-9bebc600-2a81-11eb-85c0-ab85e6f2802b.png">

Reply:
@staabm nach unserer diskussion letztens über psalm vs phpstan: Würdest du denn eigentlich hier bei redaxo phpstan drin lassen wollen?
Denn ohne hätte sich dieses Problem ja zum beispiel erübrigt, dank psalm-scope-this für psalm.

Reply:
An den aufrufenden stellen (in fragmenten) kann phpstorm jetzt die methoden sehen auf grund der zeilen die es hier anmeckert. 

Ich würde die errors an dieser stelle ignorieren.. im prinzip ist es schon doppelt gemoppelt aber dafür gehts überall :)

Reply:
Ich würde phpstan gerne drinnen lassen.  Ich finde die tools ergänzen sich

Reply:
Mich stört auch etwas, dass die Datei überall nun rot angezeigt wird:

<img width="928" alt="Bildschirmfoto 2020-11-19 um 16 13 08" src="https://user-images.githubusercontent.com/330436/99684812-29c7b100-2a82-11eb-8c35-131cd2c72dee.png">

Reply:
> An den aufrufenden stellen (in fragmenten) kann phpstorm jetzt die methoden sehen auf grund der zeilen die es hier anmeckert.

Für phpstorm reicht bereits unser `@var rex_fragement $this`, phpstorm geht mit dem scope bei `$this` da automatisch richtig um.

Wahrscheinlich können wir das Problem lösen durch eine `@noinspection ...` annotation für phpstorm (weiß die genaue notation gerade nicht auswendig).

Reply:
Scheint mir doch nicht zu gehen, da es phpstorm als harten fehler ansieht, nicht als warnungs-inspection, die man deaktivieren kann.

Reply:
Alternativ müsste man ein phpstan plugin schreiben damit er die methoden findet

Reply:
oder den scope-fehler in der phpstan-config ggf. ignorieren für die 4 methoden? 
vermutlich sagt phpstan ja, dass die methoden nicht public sind, erkennt sie aber ansonsten schon, oder?

Reply:
funktioniert es so für euch?

https://github.com/redaxo/redaxo/pull/4073